File: nomad-samples/admin/py/basic-live-channel-1.1/live_channel/get_live_channel_schedule_events.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import json, requests

def get_live_channel_schedule_events(AUTH_TOKEN, CHANNEL_ID):
    # Create header for the request
    HEADERS = {
        'Content-Type': 'application/json',
        "Authorization": "Bearer " + AUTH_TOKEN
    }

    try:
        # Send the request
        RESPONSE = requests.get(ADMIN_URL + "/liveChannel/" + CHANNEL_ID + "/liveScheduleEvent", headers= HEADERS)

        if not RESPONSE.ok:
            raise Exception()
        
        return RESPONSE.json()

    except:
        api_exception_handler(RESPONSE, "Get Live Channel Schedule Events failed")

File: nomad-samples/admin/js/basic-live-channel/live-channel/get-live-channel-schedule-events.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

/**
 * GET list of live schedule events
 *
 * Note: Schedule Events are direct calls,
 *       they are not indexed, thus, the Search API cannot be used
 *
 * @param {string} authToken    | Authorization token
 * @param {string} channelId    | The ID of the channel
 *
 * @returns {Object} containing list of live schedule events
 */
export default async function getLiveChannelScheduleEvents(authToken, channelId) {
    // Check for valid parameters
    if (!authToken || !channelId) {
        throw new Error("Get Live Schedule Events: Invalid API call");
    }

    // Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${authToken}`);

    // Send the request
    const RESPONSE = await fetch(`${prjConstants.ADMIN_API_URL}/liveChannel/${channelId}/liveScheduleEvent`, {
        method: "GET",
        headers: HEADERS
    });

    // Check for success
    if (RESPONSE && RESPONSE.ok) {
        // Return JSON response
        return await RESPONSE.json();
    }

    await apiExceptionHandler(RESPONSE, "Get Live Channel Schedule Events failed");
}
